packages/amplication-client/src/Project/ArchitectureConsole/hooks/useModelOrganizer.tsx

```diff
@@ -553,11 +555,32 @@ const useModelOrganization = ({ projectId, onMessage }: Props) => {
         (node) => node.id === sourceParentNodeId
       ).data.payload.name;
 
+      const currentTargetResource: ResourceNode = targetParent as ResourceNode;
+
       movedNodes.forEach((node) => {
-        const currentNode = currentNodes.find((n) => n.id === node.id);
+        const currentNode: EntityNode = currentNodes.find(
+          (n) => n.id === node.id
+        ) as EntityNode;
+
+        const duplicatedEntityName =
+          currentTargetResource.data.payload.entities.find(
+            (entity) =>
+              entity.displayName === currentNode.data.payload.displayName
+          );
 
         currentNode.parentNode = targetParent.id;
 
+        if (
+          duplicatedEntityName &&
+          currentNode.data.originalParentNode !== currentNode.parentNode
```

Comment:
what will happen if we move back an entity to a resource after we moved the entity to another resource and already moved another entity with the same name to that resource?

Reply:
same as above, cannot happen because we only allow a single source resource